File: src/main/java/ca/alexgirard/harpoon/SetHarpoonFileActionBase.java
package ca.alexgirard.harpoon;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public abstract class SetHarpoonFileActionBase extends AnAction {

    public abstract int getIndex();

    public SetHarpoonFileActionBase() {
        super();
    }
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        VirtualFile vFile = e.getData(PlatformDataKeys.VIRTUAL_FILE);
        HarpoonState.SetItem(vFile,getIndex(),e.getProject());
    }
}

File: src/main/kotlin/ca/alexgirard/harpoonIJ/SetHarpoonFileActionBase.kt
package ca.alexgirard.harpoonIJ

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.PlatformDataKeys

abstract class SetHarpoonFileActionBase : AnAction() {
    abstract val index: Int
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project?: return;
        val vFile = e.getData(PlatformDataKeys.VIRTUAL_FILE)
        HarpoonState.setItem(vFile, index, project)
    }
}
